src/bayesian.py

```diff
@@ -0,0 +1,50 @@
+from bayes_opt import BayesianOptimization
+import numpy as np
+from typing import List, Optional
+
+
+N_grid_points = 1000
+
+# This jupyter notebook helped me
+# https://github.com/fmfn/BayesianOptimization/blob/master/examples/visualization.ipynb
+
+
+def run_one_1d_bayesian_optimization(data: List[dict], limits: Optional[dict] = None):
+    """
+    Run one dimension bayesian optimization
+
+    :param data: observed data
+    :param limits: optional limits for the search space
+    :return: 'mu, sigma, grid, suggestion_x'
```

Comment:
I think you could probably write a sentence or so for each of these parameter descriptions, and a couple more sentences for the top-level docstring.

Ideally, someone could get a good understanding of how to use the function without reading any of the code itself.

You could talk about:
- What's the expected format of data/limits?
- What are mu/sigma/grid/suggestion_x?
- What does it mean to limit the search space? Why might I want to do that?

You could even give a high-level description of how the function works

Reply:
Also: are there any assumptions? e.g. data or limits should be ordered by x-value?